### PR TITLE
Add Nest application credentials and deprecate configuration.yaml

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add Nest application credentials and deprecate configuration.yaml and handle OOB deprecation. This is a refreshed version of https://github.com/home-assistant/home-assistant.io/pull/21303 which was the previous full config flow for nest attempt.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/core/pull/73050
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/nest.markdown
Comment (frenck) [inline source/_integrations/nest.markdown:489]:
```suggestion
- *Can’t link to Home Assistant Local Dev: Please contact Home Assistant Local Dev if the issue persists*: This typically means that the [Device Access Project](https://console.nest.google.com/device-access/project-list) is set up with an old or incorrect *OAuth Client ID*
```

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by allenporter** [comment]
We ended up handling the app auth upgrade case via config flow, so this has been updated so requesting another review for that. Parent PR is merged.

**Review by frenck — APPROVED** [review]

**Pull request merged**
